### PR TITLE
Filter search implementation

### PR DESCRIPTION
**Pull request by LuanderFarias**
I modified search.rs to let the future implementation of filters on "find a file"

### CODE REVIEW

File: core/src/api/search.rs [hunk 1]
```diff
@@ -120,6 +138,10 @@ struct FilePathFilterArgs {
 	path: Option<String>,
 	#[specta(optional)]
 	object: Option<ObjectFilterArgs>,
+	#[spectra(optional)]
```
Comment (jamiepine) [inline core/src/api/search.rs:141]: shouldn't it be "specta"?

Comment (LuanderFarias) [reply core/src/api/search.rs:141]: oh yeah, my bad

**Comment by utkubakir** [comment]
Clippy not happy

**Comment by jamiepine** [comment]
Upon a slightly closer look this isn't how we would implement this, you're directly reading a directory on the filesystem, not querying our library database. The range params are correct, but you'd have to upgrade the query in the search.paths and search.objects procedure to include those params. You wouldn't need to create a new struct for a File as we already have data structures for that too.

**Comment by LuanderFarias** [comment]
oh, ok. as i commented in the code i didn't find the path for the database and the nodes lol

**Pull request closed without merging**